packages/frontend/@n8n/chat/src/__tests__/MessageActions.spec.ts

```diff
@@ -0,0 +1,173 @@
+import type { VueWrapper } from '@vue/test-utils';
+import { mount } from '@vue/test-utils';
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+
+import { useChat } from '@n8n/chat/composables';
+import { chatEventBus } from '@n8n/chat/event-buses';
+import type { ChatMessage } from '@n8n/chat/types';
+
+import MessageActions from '../components/MessageActions.vue';
+
+vi.mock('@n8n/design-system', () => ({
+	N8nTooltip: {
+		name: 'N8nTooltip',
+		template: '<div><slot /></div>',
+	},
+	N8nIcon: {
+		name: 'N8nIcon',
+		template: '<div :icon="icon" :size="size" @click="$emit(\'click\')"></div>',
+		props: ['icon', 'size'],
+	},
+}));
+
+vi.mock('@n8n/chat/composables', () => ({
+	useChat: vi.fn(() => ({
+		sendMessage: vi.fn(),
+	})),
+	useOptions: () => ({
+		options: {
+			enableMessageActions: true,
+		},
+	}),
+}));
+
+vi.mock('@n8n/chat/event-buses', () => ({
+	chatEventBus: {
+		emit: vi.fn(),
+	},
+}));
+
+describe('MessageActions', () => {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	let wrapper: VueWrapper<any>;
```

Comment:
Same here and everywhere else in this file. In some places, we can use `mock<>` to fill in the type rather than any.